### PR TITLE
Xdr isolation and un-isolation warning on disconnected endpoint

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
related: https://github.com/demisto/etc/issues/23083
related: https://github.com/demisto/content/pull/14505

## Description
Added a new argument for both isolation and un-isolation actions in order to know if the user wants to get back an error if the endpoint is disconnected. The default is an error. Otherwise, a warning will be returned. 

## Screenshots
image
image
image

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by abaumgarten** [comment]
I will re-generate the readme file after docs review

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 1]
```diff
@@ -789,6 +789,18 @@ script:
       name: endpoint_id
       required: true
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'True'
+      description: Whether return an error when trying to isolate a disconnected endpoint.
+        When sets to false, a warnning will be returned.
+      isArray: false
+      name: disconnected_endpoint_should_return_error
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:798]:
```suggestion
      name: suppress_disconnected_endpoint_error
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 2]
```diff
@@ -789,6 +789,18 @@ script:
       name: endpoint_id
       required: true
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'True'
+      description: Whether return an error when trying to isolate a disconnected endpoint.
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:795]:
```suggestion
      description: Whether to return an error when trying to isolate a disconnected endpoint.
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 3]
```diff
@@ -805,6 +817,18 @@ script:
       name: endpoint_id
       required: true
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'True'
+      description: Whether return an error when trying to unisolate a disconnected
+        endpoint. When sets to false, a warnning will be returned.
+      isArray: false
+      name: disconnected_endpoint_should_return_error
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:826]:
```suggestion
      name: suppress_disconnected_endpoint_error
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 4]
```diff
@@ -805,6 +817,18 @@ script:
       name: endpoint_id
       required: true
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'True'
+      description: Whether return an error when trying to unisolate a disconnected
```
Comment (bakatzir) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:823]:
```suggestion
      description: Whether to return an error when trying to unisolate a disconnected
```

File: Packs/CortexXDR/ReleaseNotes/3_0_28.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Palo Alto Networks Cortex XDR - Investigation and Response
+- Added an argument for the **xdr-isolate-endpoint** and **xdr-unisolate-endpoint** commands whether or not to return an error when the endpoint is disconnected.
```
Comment (bakatzir) [inline Packs/CortexXDR/ReleaseNotes/3_0_28.md:4]:
```suggestion
- Added the *suppress_disconnected_endpoint_error* argument for the ***xdr-isolate-endpoint*** and the ***xdr-unisolate-endpoint*** commands which dictates whether to return an error when the endpoint is disconnected.
```

**Review by bakatzir — COMMENTED** [review]
please consider renaming the arg to the suggested. I did not review the code.

**Comment by abaumgarten** [comment]
@yuvalbenshalom can you review this, please?

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 6]
```diff
@@ -789,6 +789,18 @@ script:
       name: endpoint_id
       required: true
       secret: false
+    - auto: PREDEFINED
+      default: false
+      defaultValue: 'True'
```
Comment (yuvalbenshalom) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:794]:
```suggestion
      defaultValue: 'true'
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml [hunk 7]
```diff
@@ -805,6 +817,18 @@ script:
...
```
Comment (yuvalbenshalom) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.yml:822]:
```suggestion
      defaultValue: 'true'
```

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 8]
```diff
@@ -1881,7 +1881,7 @@ def insert_cef_alerts_command(client, args):
 
 def isolate_endpoint_command(client, args):
     endpoint_id = args.get('endpoint_id')
-
+    disconnected_should_return_error = argToBoolean(args.get('suppress_disconnected_endpoint_error', True))
```
Comment (yuvalbenshalom) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1884]:
```suggestion
    disconnected_should_return_error = not argToBoolean(args.get('suppress_disconnected_endpoint_error', False))
```
while it is correct that if `suppress_disconnected_endpoint_error` does not exist, then `disconnected_should_return_error` should be true, 
if `uppress_disconnected_endpoint_error` exists, `disconnected_should_return_error` should get the opposite value of it.

File: Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py [hunk 9]
```diff
@@ -1922,7 +1928,7 @@ def isolate_endpoint_command(client, args):
 
 def unisolate_endpoint_command(client, args):
     endpoint_id = args.get('endpoint_id')
-
+    disconnected_should_return_error = argToBoolean(args.get('suppress_disconnected_endpoint_error', True))
```
Comment (yuvalbenshalom) [inline Packs/CortexXDR/Integrations/CortexXDRIR/CortexXDRIR.py:1931]:
```suggestion
    disconnected_should_return_error = not argToBoolean(args.get('suppress_disconnected_endpoint_error', False))
```

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
see my comments

**Comment by abaumgarten** [comment]
@yuvalbenshalom @bakatzir I think we might have a misunderstanding here.
@bakatzir requested to change the new argument name to `suppress_disconnected_endpoint_error`, but then the description is incorrect. When sets to True, an error shouldn’t be raised, When sets to False an error should be raised (the default behaviour).
Then, @yuvalbenshalom's review does not hold after this change, please correct me if I am wrong. Maybe it is worth changing the argument name? 
thanks

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**